api/nodes.go

```diff
@@ -665,6 +666,38 @@ type StatValue struct {
 	Desc string `json:",omitempty"`
 }
 
+func (v *StatValue) String() string {
+	switch {
+	case v.BoolVal != nil:
+		return strconv.FormatBool(*v.BoolVal)
+	case v.StringVal != nil:
+		return *v.StringVal
+	case v.FloatNumeratorVal != nil:
+		str := strconv.FormatFloat(*v.FloatNumeratorVal, 'f', -1, 64)
```

Comment:
curious if we tested the magic "-1" option vs setting it to something small like 3.  What does it do with values like 1.000000003? I am asking because this may impact the CLI with unnecessary wrapping of long floats, I noticed this with my recent CLI work for displaying floating point scores.

Reply:
I experimented with small values while development but didn't like the feel of them - as the precision is strict so would have values like `1.500 / 2.000`, instead of `1.5 / 2` which feels more natural.

Rather setting format and precision arbitrarily, I decided to leave it up to the driver to report the necessary precision and we'll display it as is.

That being said, it may be nice to have consistent approach to floats.

Reply:
I used %3g for affinities, and didn't see it do `1.500`. examples -  https://play.golang.org/p/JaDHNc0eE_0

Reply:
TIL about `g` - awesome thanks!  Looks like formatting with `g` behaves differently from `f`: https://play.golang.org/p/eNj3YuzfqFi.  Thanks, I'll follow up with a new PR.

Reply:
Actually with `%g` the precision controls all digits not just decimal places:

```
fmt.Printf("%.3g", 1323.0) // prints `1.32e+03`
```

what do you think?

Reply:
`1,32e+03` isn't ideal. For affinities the values were either 0, 1, or a floating point between 0 and 1 so %g worked out well. For more general purpose stats its probably better to round it to 3 digits of precision first and then do a format print? Something like `strconv.FormatFloat(math.Floor(x*1000)/1000), 'f', -1, 64)`?